### PR TITLE
chore: update/improve the structure of the scripts so they are more intuitive

### PR DESCRIPTION
**Pull request by markmeliaa**
The scripts were quite a mess and there was some functionality that got mixed and other functionality that could be separated into a different script, so that's what this is about.

Bix now has:
- `PlayerManagerScript` to manage her stats and any other functionalities in the future.
- `InteractWithPowerUpScript` to manage the interactions between her and the powerups.
- `PlayerDeathManager` to manage everything related to her death, what should happen when her life hits 0 (this was inside the `HealthSystem` script, which made that script a bit confusing.

The drones (and any future other enemies) now have:
- `EnemyDeathScript`, same as the script that manages Bix death, functionality that was inside the `HealthSystem` script that was divided into another script. This script also manages the request of a powerup that an enemy could drop after its death.

The powerups now have:
- `PowerUpLogicScript`, practically the same as the previous script (`PowerUpScript`) but with a more intuitive name and some code logic changed.

And a new script `PowerUpsManagerScript` that will act as the manager of the powerups in the current scene (just set this script in an empty gameobject and that's all, it does not need to be attached to any GameObject in particular)

### CODE REVIEW

File: Game/Scripts/PowerUpsManagerScript.cpp [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+{
+	return savedPowerUp;
+}
+
+const PowerUpType& PowerUpsManagerScript::GetActivePowerUpType() const
+{
+	return activePowerUp;
+}
+
+void PowerUpsManagerScript::EliminateCurrentPowerUpEffect()
```
Comment (DADSILVER) [inline Game/Scripts/PowerUpsManagerScript.cpp:98]: In this func activePowerUp is set to NONE in the init so don't go to enter in the others ifs.

**Review by DADSILVER — APPROVED** [review]
Good job, Only one thing.

File: Game/Scripts/PowerUpsManagerScript.cpp (hunk 1 above)
Comment (markmeliaa) [reply Game/Scripts/PowerUpsManagerScript.cpp:99]: Good catch, thanks!

**Review by PaablooCH — CHANGES_REQUESTED** [review]
👍, few things:

File: Game/Scripts/PlayerDeathScript.cpp [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+{
+	componentAnimation = owner->GetComponent<ComponentAnimation>();
+}
+
+void PlayerDeathScript::ManagePlayerDeath() const
+{
+	DisablePlayerActions();
+
+#ifndef ENGINE
+	if (loseSceneName != "" && !componentAnimation->isPlaying() && componentAnimation->GetActualStateName() == "Death")
```
[the remark this answers is not in the document]
Comment (markmeliaa) [reply Game/Scripts/PlayerDeathScript.cpp:29]: Not related to this pr

File: Game/Scripts/BixAttackScript.cpp [hunk 3]
```diff
@@ -139,17 +138,16 @@ void BixAttackScript::CheckCollision()
 		if (Physics::Raycast(line, hit, transform->GetOwner()))
 		{
 			playSFX = true;
-			if (hit.gameObject->GetRootGO())
+			if (hit.gameObject->GetRootGO() && hit.gameObject->GetRootGO()->CompareTag("Enemy"))
```
[the remark this answers is not in the document]
Comment (markmeliaa) [reply Game/Scripts/BixAttackScript.cpp:141]: I think that's not related to this pr, but what do you mean?

**Pull request merged**